### PR TITLE
Project week 3

### PR DESCRIPTION
**Pull request by SusanaPerezJ**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -30,32 +30,32 @@
...
-      Comments comments = new Comments(id, comment, timestamp);
-      commentList.add(comments);
+    int maxCommentNum = getMaxComments(request);
+    if(maxCommentNum == -1){
+        System.out.println("could not convert to int");
     }
 
+    final List<Comments> commentList = new ArrayList<Comments>();
+    for (Entity entity : results.asIterable()) {
+        if(commentList.size() < maxCommentNum){
```
Comment (wait321) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:51]: This works, but it loops through the entire list of results every time. More efficient if you just do
for(Entity entity: results...) {
  if(commentList.size() >= maxCommentNum) {
    break;
  }
  ...
}

The `break` will exit the for loop

File: portfolio/src/main/webapp/index.html [hunk 2]
```diff
@@ -39,10 +39,15 @@ <h2>Random Facts about me</h2>
         <textarea name="comment-box" rows="6" cols="100"></textarea>
         <button type="submit">Submit comment</button>   
       </form> 
-      <p>Comments:</p>
+      <form class="space-between-elements" onsubmit="getCommentThread(); return false;">
```
Comment (wait321) [inline portfolio/src/main/webapp/index.html:42]: This works, but it's not that great. You're submitting the form and then overriding the submit behavior to call getCommentThread() and then stop the submit by returning false. All of your behavior is in the method, so you don't need the form submit at all. You could just leave out the `form` altogether and just replace your submit button with `<button type="button" onclick="getCommentThread()">..` which will just call the method directly without all of the form submit behavior.

Basically, what you did that for the delete button

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -53,6 +55,23 @@
     const titleElement = document.createElement('span');
     titleElement.innerText = text.comment;
 
+    const deleteButtonElement = document.createElement('button');
+    deleteButtonElement.innerText = 'Delete';
+    deleteButtonElement.addEventListener('click', () => {
+      deleteComment(text);
```
Comment (wait321) [inline portfolio/src/main/webapp/script.js:61]: I was looking at this wondering why passing in the comment text would work, then I read the entire file and realized that this is a Comment object. If you named this `commentJson` it would be easy to figure out what's happening just looking at this part by itself

**Review by wait321 — APPROVED** [review]
Looks good overall. Some optional comments to improve future reviews
